pkgs/by-name/vo/voicevox-engine/package.nix

```
@@ -0,0 +1,106 @@
+{
+  lib,
+  fetchFromGitHub,
+  python3,
+  replaceVars,
+  voicevox-core,
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "voicevox-engine";
+  version = "0.20.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "VOICEVOX";
+    repo = "voicevox_engine";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Gib5R7oleg+XXyu2V65EqrflQ1oiAR7a09a0MFhSITc=";
+  };
+
+  patches = [
+    # the package uses poetry for dependency management, but not package definition
+    # this patch makes the package installable via poetry-core
+    (replaceVars ./make-installable.patch {
+      inherit version;
+    })
+  ];
+
+  build-system = with python3.pkgs; [
+    poetry-core
+  ];
+
+  dependencies = with python3.pkgs; [
+    numpy
+    fastapi
+    jinja2
+    python-multipart
+    uvicorn
+    soundfile
+    pyyaml
+    pyworld
+    passthru.pyopenjtalk
+    semver
+    platformdirs
+    soxr
+    pydantic
+    starlette
+  ];
```

Because of nasty `with`, when using with to factor a list, ensure that each element is part of the factored element.

```suggestion
  dependencies =  [ passthru.pyopenjtalk ] ++ (with python3Packages; [
    numpy
    fastapi
    jinja2
    python-multipart
    uvicorn
    soundfile
    pyyaml
    pyworld
    semver
    platformdirs
    soxr
    pydantic
    starlette
  ]);
```